src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -396,10 +396,18 @@ CATCH_RETURN()
         std::vector<DWRITE_SHAPING_TEXT_PROPERTIES> textProps(textLength);
         std::vector<DWRITE_SHAPING_GLYPH_PROPERTIES> glyphProps(maxGlyphCount);
 
+        // Get the features to apply to the font
+        auto features = _fontRenderData->DefaultFontFeatures();
+        DWRITE_FONT_FEATURE* featureList = features.data();

```

Comment:
Concern: if the user specifies no features, are we going to pass `{null,0}` (typographicFeatures) down into DWrite when the text _is_ complex?

Will that disable mkmk rclt ccmp ...?

Do we need to pass _null_ on line R423?

Reply:
actually yeah `featureList` won't be null (!) because vector doesn't work that way. It will return a pointer to its internal storage, and we will say "here you go DWrite, ZERO FEATURES."

Reply:
Unless this list will always contain the default features?

Reply:
This list will always contain the default features. 

The reason for this is that we _always_ call `SetFeatures` when `UpdateFont` gets called, and `SetFeatures` will reinitialize the feature vector (with the default features), and then add the ones provided by the user. We have to do it this way because we have no way of knowing whether the user deleted features, so we always need to clear the vector and reinitialize it

Reply:
so just so I understand - the list will always contain the default features, and it will always be used if:

1. the text is complex, OR
2. the user wants special features

It will not be used, even though it is populated, if the text is simple and the user does not want special features?

Reply:
Yes that's correct!